lib/web_ui/lib/src/engine/text_editing.dart

```diff
@@ -888,3 +900,45 @@ class _EditableSizeAndTransform {
   final double height;
   final Float64List transform;
 }
+
+/// Used for detecting the changes in the selection.
+///
+/// Currently only used in Firefox.
+///
+/// In Firefox, when cursor moves, nor selectionChange neither onInput
+/// events are triggered. We are listening to keyup event. Selection start,
+/// end values are used to decide if the cursor moved.
+///
+/// Specific keycodes are not checked since users/applicatins can bind their own
+/// keys to move the cursor.
+class SelectionChangeDetection {
+  int _start = 0;
+  int _end = 0;
+
+  /// Method which decides if the selection has changed (cursor moved) compared
+  /// to the previous values.
+  ///
+  /// After each keyup, the start/end values of the selection is compared to the
+  /// previously saved start/end values.
+  bool hasChangeDetected(html.HtmlElement domElement) {
```

Comment:
nit: how about `detectChange`? It's not a getter and it has side effects, and methods are expected to sound imperative.

Reply:
done.